modules/data.atmosphere/R/check_met_input.R

```diff
@@ -0,0 +1,97 @@
+#' Check a meteorology data file for compliance with the PEcAn standard
+#'
+#' @param metfile Path of met file to check, as a scalar character.
+#' @param variable_table `data.frame` linking standard names to their
+#'   units. Defaults to [pecan_standard_met_table].
+#' @return `TRUE` (invisibly) if successful, otherwise throw an informative error.
+#' @author Alexey Shiklomanov
+#' @export
+check_met_input_file <- function(metfile, variable_table = pecan_standard_met_table) {
+
+  metfile <- normalizePath(metfile)
+
+  PEcAn.logger::severeifnot(
+    file.exists(metfile),
+    msg = glue::glue("File '{metfile}' does not exist.")
+  )
+
+  nc <- ncdf4::nc_open(metfile)
+
+  test_dims <- tryCatch(
+    testthat::test_that(
+      "Dimensions are correct.",
+      {
+        dimensions <- nc[["dim"]]
+        testthat::expect_type(dimensions, "list")
+        testthat::expect_equal(length(dimensions), 3)
+        testthat::expect_true("time" %in% names(dimensions))
+        testthat::expect_match(
+          ncdf4::ncatt_get(nc, "time", "units")[["value"]],
+          "^days since [[:digit:]]{4}-[[:digit:]]{2}-[[:digit:]]{2}T[[:digit:]]{2}:[[:digit:]]{2}:[[:digit:]]{2}Z$"
+        )
+         testthat::expect_true("latitude" %in% names(dimensions))
+         testthat::expect_equal(
+           ncdf4::ncatt_get(nc, "latitude", "units")[["value"]],
+           "degrees_north"
+         )
+         testthat::expect_true("longitude" %in% names(dimensions))
```

Comment:
Do we want to enforce dimension order? If so, all three of these could be bundled to `expect_equal(c("longitude", "latitude", "time"), names(dimensions))`

Reply:
I'm not actually sure if the order of dimensions in `nc[["dim"]]` matters. I _think_ (and I may be wrong) that it can be different for every variable, and _there_ we _should_ actually check that the dimension order is correct because it affects interpretation of the grid (e.g. `time` in the first vs. last dimension). But I very well may be wrong about that.

Reply:
Good catch, I was conflating those.